core/frontend/src/Viewport.ts

```diff
@@ -2583,17 +2583,23 @@ export abstract class Viewport implements IDisposable, TileUser {
       haveExternalTexRequests = system.hasExternalTextureRequests;
       haveNewTiles = !this.areAllTileTreesLoaded || this._hasMissingTiles;
       if (!haveNewTiles) {
-        // ViewAttachments and 3d section drawing attachments render to separate off-screen viewports - check those too.
-        for (const vp of this.view.secondaryViewports) {
-          if (vp.numRequestedTiles > 0) {
-            haveNewTiles = true;
-            break;
-          }
+        const tilesThisVp = IModelApp.tileAdmin.getTilesForUser(this);
```

Comment:
Would be nice to consolidate code copied between the two implementations of `waitForSceneCompletion` rather than adding more, if possible.